packages/wxt/src/core/utils/manifest.ts

```diff
@@ -467,7 +467,7 @@ function discoverIcons(
 }
 
 function addDevModeCsp(manifest: Manifest.WebExtensionManifest): void {
-  const permission = `http://${wxt.server?.hostname ?? ''}/*`;
+  const permission = `${wxt.server?.origin.replace(/:[0-9]+$/, '') ?? `http://localhost:${wxt.server?.port}`}/*`;
```

Comment:
The port has to be removed from `origin` in order to comply with existing CSP rules tests.  
I think changing the expected CSP is out of scope for this PR. If I encounter issues using the current rules with devcontainers I'll open another PR!